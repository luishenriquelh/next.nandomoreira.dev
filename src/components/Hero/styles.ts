import styled from 'styled-components';

export const Hero = styled.section`
  align-items: center;
  background-color: #272643;
  display: flex;
  font-size: 2rem;
  height: 100%;
  justify-content: center;
  min-height: 100vh;
  width: 100vw;
`;

export const HeroInner = styled.div`
  color: #f6fafd;
  text-align: center;
`;
