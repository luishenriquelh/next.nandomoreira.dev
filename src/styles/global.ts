import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  html {
    font-size: 16px;
    box-sizing: border-box;
  }

  * {
    margin: 0;
    padding: 0;
    box-sizing: inherit;
  }

  html, body, #__next {
    height: 100%;
  }

  body {
    background-color: ${({ theme }) => theme?.backgroundColor};
    color: ${({ theme }) => theme?.textColor};
    font-family: ${({ theme }) => theme?.fontFamily};
    font-size: ${({ theme }) => theme?.fontSize};
    line-height: ${({ theme }) => theme?.lineHeight};
    font-weight: ${({ theme }) => theme?.fontWeight?.medium};
  }

  h1, h2, h3, h4 {
    font-family: ${({ theme }) => theme?.fontFamilyHeading};
    font-weight: ${({ theme }) => theme?.fontWeight?.bold};
  }
`;
