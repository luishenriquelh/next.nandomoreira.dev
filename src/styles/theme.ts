const fontFamilyBase = `-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu,
  Cantarell, "Open Sans", "Helvetica Neue", sans-serif`;

const defaultTheme = {
  fontFamily: fontFamilyBase,
  fontFamilyHeading: `"Montserrat", ${fontFamilyBase}`,
  fontSize: `1rem`,
  lineHeight: `1.4`,
  fontWeight: {
    medium: 500,
    semiBold: 600,
    bold: 800,
  },
};

const darkColors = {
  backgroundColor: `#272643`,
  borderColor: `#f6fafd`,
  textColor: `#f6fafd`,
  logo: {
    pathColor: `#e5e5e5`,
    rectColor: `#f6fafd`,
  },
};

const lightColors = {
  backgroundColor: `#f6fafd`,
  borderColor: `#2c698d`,
  textColor: `#272643`,
  logo: {
    pathColor: `#272643`,
    rectColor: `#2c698d`,
  },
};

export const theme = {
  dark: { ...defaultTheme, ...darkColors },
  light: { ...defaultTheme, ...lightColors },
};
