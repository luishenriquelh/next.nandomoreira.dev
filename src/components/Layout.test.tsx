import React from 'react';
import { createRouter } from 'next/router';
import { render } from '@testing-library/react';
import { RouterContext } from 'next/dist/next-server/lib/router-context';
import { theme } from '@nandomoreira/styles';
import { Layout } from '.';

const router = createRouter('/', {}, '/', {
  initialProps: {},
  pageLoader: jest.fn(),
  App: jest.fn(),
  Component: jest.fn(),
});

describe('<Layout />', () => {
  test('Render component with dark theme', () => {
    const componentTree = render(
      <RouterContext.Provider value={router}>
        <Layout theme={theme.dark}>Hello Layout</Layout>
      </RouterContext.Provider>,
    );
    expect(componentTree).toMatchSnapshot();
  });

  test('Render component with light theme', () => {
    const componentTree = render(
      <RouterContext.Provider value={router}>
        <Layout theme={theme.light}>Hello Layout</Layout>
      </RouterContext.Provider>,
    );
    expect(componentTree).toMatchSnapshot();
  });
});
