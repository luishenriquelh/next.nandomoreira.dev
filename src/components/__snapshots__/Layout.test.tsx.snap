// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Render component with dark theme 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border: 0;
}

.c5.--focusable:focus,
.c5.--focusable:focus-within {
  position: static;
  width: auto;
  height: auto;
  padding: 0.5rem;
  margin: 0.25rem;
  overflow: visible;
  -webkit-clip: auto;
  clip: auto;
  white-space: normal;
}

.c0 {
  background-color: #272643;
  border: 0;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  text-align: center;
  top: 40px;
  left: 40px;
  white-space: nowrap;
  width: 1px;
  z-index: 10001;
}

.c0 a {
  width: 100%;
  color: #f6fafd;
  display: inline-block;
}

.c0:focus,
.c0:focus-within {
  -webkit-clip: auto;
  clip: auto;
  display: block;
  height: auto;
  margin: 0;
  overflow: visible;
  padding: 0.5rem;
  position: fixed;
  white-space: normal;
  width: auto;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none transparent;
  border: none;
  bottom: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: inline-block;
  font-size: 16px;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 40px;
  padding: 0;
  position: fixed;
  right: 40px;
  width: 40px;
}

.c4 {
  cursor: pointer;
  display: inline-block;
}

.c6 path {
  fill: #e5e5e5;
}

.c6 rect {
  fill: #f6fafd;
}

.c1 {
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  z-index: 1000;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 2.5rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  color: #f6fafd;
  cursor: pointer;
  display: inline-block;
  font-family: "Montserrat",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu, Cantarell,"Open Sans","Helvetica Neue",sans-serif;
  font-size: 1rem;
  font-weight: 500;
  margin: 0 10px;
  padding: 5px 10px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8::before,
.c8::after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 2px;
  width: 0%;
  background-color: #f6fafd;
  opacity: 0.5;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c8::before {
  left: 10px;
}

.c8::after {
  right: 10px;
}

.c8.--active::before,
.c8:active::before,
.c8:hover::before,
.c8.--active::after,
.c8:active::after,
.c8:hover::after {
  width: calc(50% - 10px);
}

@media (min-width:768px) {
  .c1 {
    position: fixed;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <a
          href="#mainContent"
        >
          Ir para o conteúdo principal
        </a>
      </div>
      <header
        class="c1"
      >
        <div
          class="c2"
        >
          <h1
            class="c3"
          >
            <a
              class="c4"
              href="/"
            >
              <span
                class="c5"
              >
                undefinedundefined
              </span>
              <svg
                class="c6"
                fill="none"
                height="45"
                width="50"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  undefinedundefined
                </title>
                <path
                  d="M43.247 0v45H32.662L12.597 21.021V45H0V0h10.584L30.65 23.979V0h12.598z"
                />
                <rect
                  class="dot"
                  height="12.857"
                  rx="6.429"
                  width="12.987"
                  x="37.013"
                  y="32.143"
                />
              </svg>
            </a>
          </h1>
          <nav
            class="c7"
          >
            <a
              class="c8"
              href="/sobre"
            >
              Sobre
            </a>
            <a
              class="c8"
              href="/portfolio"
            >
              Portfolio
            </a>
            <a
              class="c8"
              href="/contato"
            >
              Contato
            </a>
            <a
              class="c8"
              href="/cv"
            >
              Resume
            </a>
            <a
              class="c8"
              href="https://blog.nandomoreira.dev"
              rel="noopener noreferrer"
              target="_blank"
            >
              Blog
            </a>
          </nav>
        </div>
      </header>
      <main
        id="mainContent"
      >
        Hello Layout
      </main>
      <footer>
        <button
          class="c9"
        >
          <svg
            fill="none"
            height="36"
            width="36"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.113 19.567c-1.016.27-2.064.407-3.115.407-3.203 0-6.21-1.245-8.47-3.504a12.038 12.038 0 01-3.095-11.584 1.5 1.5 0 00-1.838-1.835 15.02 15.02 0 00-6.733 3.873c-5.847 5.847-5.847 15.364 0 21.215a14.904 14.904 0 0010.608 4.394 14.893 14.893 0 0010.605-4.393 15.011 15.011 0 003.874-6.736 1.502 1.502 0 00-1.836-1.837zm-4.158 6.452a11.92 11.92 0 01-8.484 3.515 11.93 11.93 0 01-8.487-3.517c-4.677-4.678-4.677-12.292 0-16.97a11.884 11.884 0 013.09-2.225 15.04 15.04 0 004.335 11.772 14.956 14.956 0 0011.772 4.336 12.055 12.055 0 01-2.226 3.089z"
              fill="#272643"
            />
          </svg>
        </button>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="Skippystyles__Skippy-becy8g-0 isIQwu"
    >
      <a
        href="#mainContent"
      >
        Ir para o conteúdo principal
      </a>
    </div>
    <header
      class="Headerstyles__Header-sc-1oalovf-0 isOaYz"
    >
      <div
        class="Headerstyles__HeaderInner-sc-1oalovf-1 hVIASu"
      >
        <h1
          class="Headerstyles__Logo-sc-1oalovf-2 kmbipE"
        >
          <a
            class="Logostyles__Logo-ztqmk5-0 jbrqWl"
            href="/"
          >
            <span
              class="SrOnlystyles__Tag-sc-1xgrgn4-0 gepsbO"
            >
              undefinedundefined
            </span>
            <svg
              class="Logostyles__Svg-ztqmk5-1 nSBRD"
              fill="none"
              height="45"
              width="50"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                undefinedundefined
              </title>
              <path
                d="M43.247 0v45H32.662L12.597 21.021V45H0V0h10.584L30.65 23.979V0h12.598z"
              />
              <rect
                class="dot"
                height="12.857"
                rx="6.429"
                width="12.987"
                x="37.013"
                y="32.143"
              />
            </svg>
          </a>
        </h1>
        <nav
          class="Headerstyles__Nav-sc-1oalovf-3 cOClTf"
        >
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 bWVbqa"
            href="/sobre"
          >
            Sobre
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 bWVbqa"
            href="/portfolio"
          >
            Portfolio
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 bWVbqa"
            href="/contato"
          >
            Contato
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 bWVbqa"
            href="/cv"
          >
            Resume
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 bWVbqa"
            href="https://blog.nandomoreira.dev"
            rel="noopener noreferrer"
            target="_blank"
          >
            Blog
          </a>
        </nav>
      </div>
    </header>
    <main
      id="mainContent"
    >
      Hello Layout
    </main>
    <footer>
      <button
        class="DarkModestyles__DarkMode-sc-12towz7-0 kLOegO"
      >
        <svg
          fill="none"
          height="36"
          width="36"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.113 19.567c-1.016.27-2.064.407-3.115.407-3.203 0-6.21-1.245-8.47-3.504a12.038 12.038 0 01-3.095-11.584 1.5 1.5 0 00-1.838-1.835 15.02 15.02 0 00-6.733 3.873c-5.847 5.847-5.847 15.364 0 21.215a14.904 14.904 0 0010.608 4.394 14.893 14.893 0 0010.605-4.393 15.011 15.011 0 003.874-6.736 1.502 1.502 0 00-1.836-1.837zm-4.158 6.452a11.92 11.92 0 01-8.484 3.515 11.93 11.93 0 01-8.487-3.517c-4.677-4.678-4.677-12.292 0-16.97a11.884 11.884 0 013.09-2.225 15.04 15.04 0 004.335 11.772 14.956 14.956 0 0011.772 4.336 12.055 12.055 0 01-2.226 3.089z"
            fill="#272643"
          />
        </svg>
      </button>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Layout /> Render component with light theme 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c5 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border: 0;
}

.c5.--focusable:focus,
.c5.--focusable:focus-within {
  position: static;
  width: auto;
  height: auto;
  padding: 0.5rem;
  margin: 0.25rem;
  overflow: visible;
  -webkit-clip: auto;
  clip: auto;
  white-space: normal;
}

.c0 {
  background-color: #f6fafd;
  border: 0;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  text-align: center;
  top: 40px;
  left: 40px;
  white-space: nowrap;
  width: 1px;
  z-index: 10001;
}

.c0 a {
  width: 100%;
  color: #272643;
  display: inline-block;
}

.c0:focus,
.c0:focus-within {
  -webkit-clip: auto;
  clip: auto;
  display: block;
  height: auto;
  margin: 0;
  overflow: visible;
  padding: 0.5rem;
  position: fixed;
  white-space: normal;
  width: auto;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none transparent;
  border: none;
  bottom: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: inline-block;
  font-size: 16px;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 40px;
  padding: 0;
  position: fixed;
  right: 40px;
  width: 40px;
}

.c4 {
  cursor: pointer;
  display: inline-block;
}

.c6 path {
  fill: #272643;
}

.c6 rect {
  fill: #2c698d;
}

.c1 {
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  z-index: 1000;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 2.5rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  color: #272643;
  cursor: pointer;
  display: inline-block;
  font-family: "Montserrat",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu, Cantarell,"Open Sans","Helvetica Neue",sans-serif;
  font-size: 1rem;
  font-weight: 500;
  margin: 0 10px;
  padding: 5px 10px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8::before,
.c8::after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 2px;
  width: 0%;
  background-color: #272643;
  opacity: 0.5;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c8::before {
  left: 10px;
}

.c8::after {
  right: 10px;
}

.c8.--active::before,
.c8:active::before,
.c8:hover::before,
.c8.--active::after,
.c8:active::after,
.c8:hover::after {
  width: calc(50% - 10px);
}

@media (min-width:768px) {
  .c1 {
    position: fixed;
  }
}

<div>
      <div
        class="c0"
      >
        <a
          href="#mainContent"
        >
          Ir para o conteúdo principal
        </a>
      </div>
      <header
        class="c1"
      >
        <div
          class="c2"
        >
          <h1
            class="c3"
          >
            <a
              class="c4"
              href="/"
            >
              <span
                class="c5"
              >
                undefinedundefined
              </span>
              <svg
                class="c6"
                fill="none"
                height="45"
                width="50"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  undefinedundefined
                </title>
                <path
                  d="M43.247 0v45H32.662L12.597 21.021V45H0V0h10.584L30.65 23.979V0h12.598z"
                />
                <rect
                  class="dot"
                  height="12.857"
                  rx="6.429"
                  width="12.987"
                  x="37.013"
                  y="32.143"
                />
              </svg>
            </a>
          </h1>
          <nav
            class="c7"
          >
            <a
              class="c8"
              href="/sobre"
            >
              Sobre
            </a>
            <a
              class="c8"
              href="/portfolio"
            >
              Portfolio
            </a>
            <a
              class="c8"
              href="/contato"
            >
              Contato
            </a>
            <a
              class="c8"
              href="/cv"
            >
              Resume
            </a>
            <a
              class="c8"
              href="https://blog.nandomoreira.dev"
              rel="noopener noreferrer"
              target="_blank"
            >
              Blog
            </a>
          </nav>
        </div>
      </header>
      <main
        id="mainContent"
      >
        Hello Layout
      </main>
      <footer>
        <button
          class="c9"
        >
          <svg
            fill="none"
            height="36"
            width="36"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.113 19.567c-1.016.27-2.064.407-3.115.407-3.203 0-6.21-1.245-8.47-3.504a12.038 12.038 0 01-3.095-11.584 1.5 1.5 0 00-1.838-1.835 15.02 15.02 0 00-6.733 3.873c-5.847 5.847-5.847 15.364 0 21.215a14.904 14.904 0 0010.608 4.394 14.893 14.893 0 0010.605-4.393 15.011 15.011 0 003.874-6.736 1.502 1.502 0 00-1.836-1.837zm-4.158 6.452a11.92 11.92 0 01-8.484 3.515 11.93 11.93 0 01-8.487-3.517c-4.677-4.678-4.677-12.292 0-16.97a11.884 11.884 0 013.09-2.225 15.04 15.04 0 004.335 11.772 14.956 14.956 0 0011.772 4.336 12.055 12.055 0 01-2.226 3.089z"
              fill="#272643"
            />
          </svg>
        </button>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="Skippystyles__Skippy-becy8g-0 bIMpri"
    >
      <a
        href="#mainContent"
      >
        Ir para o conteúdo principal
      </a>
    </div>
    <header
      class="Headerstyles__Header-sc-1oalovf-0 isOaYz"
    >
      <div
        class="Headerstyles__HeaderInner-sc-1oalovf-1 hVIASu"
      >
        <h1
          class="Headerstyles__Logo-sc-1oalovf-2 kmbipE"
        >
          <a
            class="Logostyles__Logo-ztqmk5-0 jbrqWl"
            href="/"
          >
            <span
              class="SrOnlystyles__Tag-sc-1xgrgn4-0 gepsbO"
            >
              undefinedundefined
            </span>
            <svg
              class="Logostyles__Svg-ztqmk5-1 jmMqlm"
              fill="none"
              height="45"
              width="50"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                undefinedundefined
              </title>
              <path
                d="M43.247 0v45H32.662L12.597 21.021V45H0V0h10.584L30.65 23.979V0h12.598z"
              />
              <rect
                class="dot"
                height="12.857"
                rx="6.429"
                width="12.987"
                x="37.013"
                y="32.143"
              />
            </svg>
          </a>
        </h1>
        <nav
          class="Headerstyles__Nav-sc-1oalovf-3 cOClTf"
        >
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 etxoFe"
            href="/sobre"
          >
            Sobre
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 etxoFe"
            href="/portfolio"
          >
            Portfolio
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 etxoFe"
            href="/contato"
          >
            Contato
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 etxoFe"
            href="/cv"
          >
            Resume
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 etxoFe"
            href="https://blog.nandomoreira.dev"
            rel="noopener noreferrer"
            target="_blank"
          >
            Blog
          </a>
        </nav>
      </div>
    </header>
    <main
      id="mainContent"
    >
      Hello Layout
    </main>
    <footer>
      <button
        class="DarkModestyles__DarkMode-sc-12towz7-0 kLOegO"
      >
        <svg
          fill="none"
          height="36"
          width="36"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.113 19.567c-1.016.27-2.064.407-3.115.407-3.203 0-6.21-1.245-8.47-3.504a12.038 12.038 0 01-3.095-11.584 1.5 1.5 0 00-1.838-1.835 15.02 15.02 0 00-6.733 3.873c-5.847 5.847-5.847 15.364 0 21.215a14.904 14.904 0 0010.608 4.394 14.893 14.893 0 0010.605-4.393 15.011 15.011 0 003.874-6.736 1.502 1.502 0 00-1.836-1.837zm-4.158 6.452a11.92 11.92 0 01-8.484 3.515 11.93 11.93 0 01-8.487-3.517c-4.677-4.678-4.677-12.292 0-16.97a11.884 11.884 0 013.09-2.225 15.04 15.04 0 004.335 11.772 14.956 14.956 0 0011.772 4.336 12.055 12.055 0 01-2.226 3.089z"
            fill="#272643"
          />
        </svg>
      </button>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
